specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkVirtualAppliance.json

```diff
@@ -1023,6 +1047,59 @@
       },
       "description": "Network Virtual Appliance NIC properties."
     },
+    "NetworkVirtualApplianceNetworkInterfaceConfiguration": {
+      "type": "object",
+      "readOnly": false,
+      "description": "Represents a single NIC configuration",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "PublicNic",
+            "PrivateNic"
+          ]
+        },
+        "properties": {
+          "$ref": "#/definitions/NetworkVirtualApplianceNetworkInterfaceConfigurationProperties"
+        }
+      }
+    },
+    "NetworkVirtualApplianceNetworkInterfaceConfigurationProperties": {
+      "type": "object",
+      "readOnly": false,
+      "description": "Represents a single NIC configuration properties",
+      "properties": {
+        "ipConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkVirtualApplianceIPConfiguration"
+          }
+        }
+      }
+    },
+    "NetworkVirtualApplianceIPConfiguration": {
+      "type": "object",
+      "readOnly": false,
+      "description": "Represents a single IP configuration",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/NetworkVirtualApplianceIPConfigurationProperties"
+        }
+      }
+    },
+    "NetworkVirtualApplianceIPConfigurationProperties": {
+      "type": "object",
+      "readOnly": false,
+      "description": "Represents a single IP configuration properties",
+      "properties": {
+        "primary": {
```

Comment:
1. lack of `description` for `primary`
2. consider using enum instead of `boolean`:https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md

Reply:
Made changes for 1st and keeping the 2nd as it is to make it consistent for primary field across networking.